client/app/components/dashboards/add-widget-dialog.html

```diff
@@ -4,46 +4,75 @@ <h4 class="modal-title">Add Widget</h4>
+          class="d-flex align-items-center justify-content-center"
+          style="position: absolute; right: 1px; top: 1px; bottom: 1px; width: 30px; background: #fff; border-radius: 3px;"
+        ><i class="text-muted fa fa-times"></i></a>
       </div>
-      <div ng-show="$ctrl.text">
-          <strong>Preview:</strong>
-          <p ng-bind-html="$ctrl.text | markdown"></p>
+    </div>
+
+    <div ng-if="!$ctrl.selectedQuery && ($ctrl.searchTerm == '')" class="scrollbox" style="max-height: 50vh">
+      <div class="list-group" ng-if="$ctrl.recentQueries.length > 0">
+        <a href="javascript:void(0)" class="list-group-item"
+          ng-repeat="query in $ctrl.recentQueries" ng-click="$ctrl.selectQuery(query.id)"
+        >
+          <span>{{query.name}}</span>
+          <span class="label label-default" ng-if="query.is_draft">Unpublished</span>
+        </a>
       </div>
-  </div>
+    </div>
 
-  <div ng-show="$ctrl.isVisualization()">
-      <div class="form-group">
-          <ui-select ng-model="$ctrl.query.selected" theme="bootstrap" reset-search-input="false" on-select="$ctrl.onQuerySelect($item, $model)">
-              <ui-select-match placeholder="Search a query by name">{{$select.selected.name}}</ui-select-match>
-              <ui-select-choices repeat="q in $ctrl.queries"
-                                 refresh="$ctrl.searchQueries($select.search)"
-                                 refresh-delay="0">
-                  <div ng-bind-html="$ctrl.trustAsHtml(q.name | highlight: $select.search)"></div>
-              </ui-select-choices>
-          </ui-select>
+    <div ng-if="!$ctrl.selectedQuery && ($ctrl.searchTerm != '')" class="scrollbox" style="max-height: 50vh">
+      <div ng-if="$ctrl.searchedQueries.length == 0" class="text-muted">
+        No results matching search term.
       </div>
+      <div class="list-group" ng-if="$ctrl.searchedQueries.length > 0">
+        <a href="javascript:void(0)" class="list-group-item"
+          ng-repeat="query in $ctrl.searchedQueries" ng-click="$ctrl.selectQuery(query.id)"
+        >
+          <span ng-bind-html="$ctrl.trustAsHtml(query.name | highlight: $ctrl.searchTerm)"></span>
+          <span class="label label-default" ng-if="query.is_draft">Unpublished</span>
+        </a>
+      </div>
+    </div>
 
-      <div ng-show="$ctrl.selected_query">
-          <div class="form-group">
-              <label for="">Choose Visualization</label>
-              <select ng-model="$ctrl.selectedVis" ng-options="vis as vis.name group by vis.type for vis in $ctrl.selected_query.visualizations" class="form-control"></select>
-          </div>
+    <div ng-show="$ctrl.selectedQuery">
+      <div class="form-group">
+        <label for="">Choose Visualization</label>
+        <select ng-model="$ctrl.selectedVis" class="form-control"
+          ng-options="vis as vis.name group by vis.type for vis in $ctrl.selectedQuery.visualizations"></select>
```

Comment:
Let's take the opportunity and make selected the visualization easier: instead of a dropdown, show small "cards" (like the ones we have for data sources/destinations) with the visualization name and type.

![image](https://user-images.githubusercontent.com/71468/36169870-2c28fa82-1106-11e8-8a4b-44c99b647426.png)

Reply:
Good idea. Should they be grouped somehow? Also I can add icons (same as in viz editor)

Reply:
In the drop down we group by type, but I don't think it's needed here.

Icons: I had the same thougth until I remembered that we have only icons for chart types, but not for top level visualization type :-( We probably should have some, but might take time until we collect for all of them.

Reply:
well, I forgot that we don't have icons for visualization types, just for chart types. So my question is not more relevant: the answer is - no any icons :-)

Reply:
:-)

Reply:
@arikfr "show small "cards" (like the ones we have for data sources/destinations) with the visualization name and type" — we tried some designs there and it brings up a few questions that are non-trivial. Like these boxes doesn't look like you need to select one while the dropdown is very trivial. I'd recommend to drop this improvement for now as it increases the scope and we should focus on v4 release.

Reply:
Here are some quick sketches to show the problem:

![Uploading Screenshot 2018-02-13 23.11.52.png…]()

Reply:
@arikfr, let's leave it for now. we discussed it with @kocsmy yesterday, and ATM we have no idea that will match all conditions good enough (we need to show viz type, viz may have long name, it should look nice both when there are 1-2 viz and a lot of viz; if showing groups - we should remember that some groups may be large and some may contain 1-2 viz etc.). we will think about it and share our ideas, but it may take some time. I fixed other notices and will update PR in a moment.

Reply:
👍